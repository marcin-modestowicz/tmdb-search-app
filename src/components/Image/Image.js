// @flow

import React from "react";

export const IMAGE_URL = "https://image.tmdb.org/t/p/";
export const IMAGE_SIZE = "w185";

type Props = {
  name: string
};

const Image = ({ name }: Props) => (
  <img src={`${IMAGE_URL}${IMAGE_SIZE}${name}`} />
);

export default Image;
