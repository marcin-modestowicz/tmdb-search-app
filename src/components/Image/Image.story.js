// @flow

import React from "react";
import { storiesOf } from "@storybook/react";
import { withKnobs, text } from "@storybook/addon-knobs";
import Image from "./Image";

storiesOf("Image", module)
  .addDecorator(withKnobs)
  .add("simple", () => (
    <Image name={text("Filename:", "/kBf3g9crrADGMc2AMAMlLBgSm2h.jpg")} />
  ));
