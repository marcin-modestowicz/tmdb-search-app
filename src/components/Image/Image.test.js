// @flow
import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Image from "./Image";

describe("Image component", () => {
  test("should render with proper src attribute value", () => {
    const image = shallow(<Image name="/kBf3g9crrADGMc2AMAMlLBgSm2h.jpg" />);

    expect(toJson(image)).toMatchSnapshot();
  });
});
