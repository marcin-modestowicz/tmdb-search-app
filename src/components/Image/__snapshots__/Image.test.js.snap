// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image component should render with proper src attribute value 1`] = `
<img
  src="https://image.tmdb.org/t/p/w185/kBf3g9crrADGMc2AMAMlLBgSm2h.jpg"
/>
`;
